pkg/jx/cmd/delete_extension.go

```diff
@@ -144,26 +146,48 @@ func (o *DeleteExtensionOptions) Run() error {
 	if len(args) == 0 {
 		return fmt.Errorf("Specify the extensions to delete using %s or delete them all using %s.", util.ColorInfo("jx delete extensions <extension name>"), util.ColorInfo("jx delete extension --all"))
 	}
-	deleteExtensions := strings.Join(args, ", ")
 
-	if !o.BatchMode && !util.Confirm(fmt.Sprintf("You are about to delete the Extensions: %s", deleteExtensions), false, "The list of Extensions to be deleted", o.In, o.Out, o.Err) {
-		return nil
+	configLookup := make(map[string]jenkinsv1.ExtensionConfig)
+	for _, c := range extensionsList.Extensions {
+		configLookup[c.Name] = c
 	}
+	extensionsToDelete := make([]jenkinsv1.Extension, 0)
 	for _, name := range args {
 		if util.StringArrayIndex(names, name) < 0 {
 			return util.InvalidOption(optionLabel, name, names)
 		}
 		ext, _ := lookup[name]
-		if ext.IsOnUninstall() {
+		children, err := extensions.GetChildrenRecursively(ext, configLookup, lookupByUUID, extensionsClient, ns, o.Verbose)
+		if err != nil {
+			return err
+		}
+		extensionsToDelete = append(extensionsToDelete, children...)
+	}
+
+	// Do a de-dupe
+	extensionsToDeleteMap := make(map[string]jenkinsv1.Extension)
+
+	for _, e := range extensionsToDelete {
+		extensionsToDeleteMap[e.Spec.UUID] = e
+	}
+
+	extensionsToDeleteStrings := make([]string, 0)
+	for _, e := range extensionsToDeleteMap {
+		extensionsToDeleteStrings = append(extensionsToDeleteStrings, e.Spec.FullyQualifiedName())
+	}
+
+	// TODO display the extensions to delete in a tree view
+	if !o.BatchMode && !util.Confirm(fmt.Sprintf("You are about to delete the Extensions: %s", strings.Join(extensionsToDeleteStrings, ", ")), false, "The list of Extensions to be deleted", o.In, o.Out, o.Err) {
```

Comment:
Do you want some kind of `-y` protection in batch mode? That would be analagous to saying 'yes' to the prompt in interactive mode

Reply:
I don't think we generally do that?

Reply:
It's an open question. If this is the correct behaviour I'm fine with it - just wanted to ask to make sure